src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,51 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  selectedTodo: Todo;
+  onClose: (todoId: number) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  onClose,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+
+  const hideModal = () => {
+    onClose(0);
+  };
+
+  const waitForUsers = async (currentTodo: Todo | null) => {
```

Comment:
selectedTodo became from props, there is no need to pass it function, just use it from closure